src/utils/task_stats_alloc/src/lib.rs

```diff
@@ -89,8 +93,90 @@ impl TaskLocalBytesAllocated {
     }
 }
 
+#[derive(Default, Debug)]
+pub struct LocalStats {
+    /// Bytes allocated in the task and also deallocated in current task.
+    cur_task_allocated: usize,
+    /// Bytes deallocated in current task and also allocated it.
+    cur_task_deallocated: usize,
+    /// Bytes deallocated in current task but allocated in other tasks.
+    other_task_deallocated: HashMap<task::Id, usize, RandomState, System>,
+    // TODO: Bytes deallocated in current task but allocated out of the runtime.
+}
+
+impl LocalStats {
+    #[inline(always)]
+    fn add(&mut self, bytes: usize) {
+        self.cur_task_allocated += bytes;
+    }
+
+    #[inline(always)]
+    fn sub(&mut self, task_id: Option<task::Id>, bytes: usize) {
+        let cur_task_id = task::id();
```

Comment:
Let the user provide a `NonZeroU64` as id? So that we can associate it with actors or tasks.